benchmarks/benchmarks.rb

```diff
@@ -1,41 +1,79 @@
 # frozen_string_literal: true
 
-require "benchmark/ips"
-
 require "tempfile"
+require "benchmark"
+require "benchmark/ips"
+require "gem_bench/jersey"
 
-github_memo_wise_path = Gem.loaded_specs["memo_wise"].full_gem_path
-
-# This string is both used for temp filepaths necessary to separate the GitHub
-# version of MemoWise and the local version, and used for the reported results
 GITHUB_MAIN = "MemoWise_GitHubMain"
-
-# We download a the main branch of MemoWise on GitHub into a tmp directory to
-# compare against the local version when we run benchmarks
-Dir.mktmpdir do |directory|
-  Dir["#{github_memo_wise_path}/lib/**/*.rb"].each do |file|
-    Tempfile.open([File.basename(file)[0..-4], ".rb"], directory) do |tempfile|
```

Comment:
Note that in GemBench, `Tempfile` would not work with some libraries, because it makes the filename unique, and thus incompatible with later `require "ny_gem/my_file"` statements.  As a result I had to switch to using `File`, but still nested inside of `Dir.mktmpdir` for automatic cleanup.  The copied and renamespaced gems are loaded into memory, but only after the copy and renamespace is complete, and then the source is deleted.  This allows much more complex gems to be loaded.